tracer/src/Datadog.Trace/OTelMetrics/OtlpMetricsSerializer.cs

```diff
@@ -0,0 +1,636 @@
+// <copyright file="OtlpMetricsSerializer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if NET6_0_OR_GREATER
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Datadog.Trace.Configuration;
+
+#nullable enable
+
+namespace Datadog.Trace.OTelMetrics
+{
+    /// <summary>
+    /// High-performance OTLP protobuf serializer that creates binary protobuf payloads
+    /// compliant with the OpenTelemetry ExportMetricsServiceRequest schema
+    /// but does not require Google.Protobuf dependency (DDSketch + OpenTelemetry patterns).
+    /// </summary>
+    internal static class OtlpMetricsSerializer
+    {
+        // Protobuf wire types
+        private const int VarInt = 0;
+        private const int Fixed64 = 1;
+        private const int LengthDelimited = 2;
+
+        private static byte[] SerializeResourceMetrics(List<MetricPoint> metrics, TracerSettings settings)
+        {
+            using var buffer = new MemoryStream();
+            using var writer = new BinaryWriter(buffer, Encoding.UTF8);
+
+            // Resource
+            var resourceData = SerializeResource(settings);
+            WriteTag(writer, FieldNumbers.Resource, LengthDelimited);
+            WriteVarInt(writer, resourceData.Length);
+            writer.Write(resourceData);
+
+            // ScopeMetrics
+            var scopeMetricsData = SerializeScopeMetrics(metrics, settings);
+            WriteTag(writer, FieldNumbers.ScopeMetrics, LengthDelimited);
+            WriteVarInt(writer, scopeMetricsData.Length);
+            writer.Write(scopeMetricsData);
+
+            return buffer.ToArray();
+        }
+
+        private static byte[] SerializeResource(TracerSettings settings)
+        {
+            using var buffer = new MemoryStream();
+            using var writer = new BinaryWriter(buffer, Encoding.UTF8);
+
+            // telemetry.sdk.name attribute (use 'opentelemetry' for ecosystem compatibility)
+            var sdkNameAttr = SerializeKeyValue("telemetry.sdk.name", "opentelemetry");
```

Comment:
We haven't discussed this yet, but I imagine `datadog` would be a good value here. Since we're not an OpenTelemetry distribution, we must not set this to `opentelemetry`.

```suggestion
            var sdkNameAttr = SerializeKeyValue("telemetry.sdk.name", "datadog");
```